crates/stages/src/stages_metrics.rs

```diff
@@ -0,0 +1,16 @@
+use metrics::{counter, increment_counter};
+use reth_interfaces::p2p::error::DownloadError;
+
+pub(crate) fn update_headers_metrics(n_headers: u64) {
```

Comment:
Should we inline(always) these calls?

Reply:
Instead of individual calls, I'd rather have a `StageMetrics` struct with `Gauge/Counter`s as fields that are mapped to the registered metrics. And this struct then encapsulates all the metrics that we have, this is easier to maintain than standalone metric functions.

I'm not sure how the exact workflow looks like in `metrics` but imo it should be possible to register them and create a Gauge/Counter for it.

Reply:
Sg

Reply:
Let me know if this sounds good:
I will implement a struct `StageMetrics` like so,
```rust
struct StageMetrics {
    pub headers_counter: Counter,
    pub headers_timeout_errors: Counter,
    pub headers_validation_errors: Counter,
    pub headers_unexpected_errors: Counter,
}
```
This struct could be initialized inside the `HeaderStage` and when a metric should be updated, e.g. the headers counter, in the code this could be done with
```
self.stage_metrics.header_counter.increment(res.len() as u64);
```
Is this consitent with your suggestions?

Reply:
This sounds right. @mattsse?

Reply:
I have refactored the metrics to encapsulate them in the `HeaderMetrics` struct now. Let me know what you think @mattsse!